stable_plugins/classical_ml/data_preparation/pandas_preprocessing/routes.py

```diff
@@ -0,0 +1,361 @@
+class FinalProcessView(MethodView):
+    """Start a long running processing task."""
+
+    @PDPreprocessing_BLP.arguments(
+        SecondInputParametersSchema(unknown=EXCLUDE), location="form"
+    )
+    @PDPreprocessing_BLP.response(HTTPStatus.OK, TaskResponseSchema())
+    @PDPreprocessing_BLP.require_jwt("jwt", optional=True)
+    def post(self, arguments, db_id: int, step_id: int):
+        """Start the calculation task."""
+        db_task: Optional[ProcessingTask] = ProcessingTask.get_by_id(id_=db_id)
+        if db_task is None:
+            msg = f"Could not load task data with id {db_id} to read parameters!"
+            TASK_LOGGER.error(msg)
+            raise KeyError(msg)
+
+        db_task.parameters = SecondInputParametersSchema().dumps(arguments)
+
+        db_task.clear_previous_step()
+        db_task.save(commit=True)
+
+        # all tasks need to know about db id to load the db entry
+        task: chain = second_task.s(
+            db_id=db_task.id, step_id=step_id
+        ) | save_task_result.s(db_id=db_task.id)
+        # save errors to db
+        task.link_error(save_task_error.s(db_id=db_task.id))
+        task.apply_async()
+
+        return redirect(
+            url_for("tasks-api.TaskView", task_id=str(db_task.id)), HTTPStatus.SEE_OTHER
+        )
+
+
+@PDPreprocessing_BLP.route("/<int:db_id>/<int:step_id>-process0/")
+class SecondProcessView(MethodView):
+    """Start a long running processing task."""
+
+    @PDPreprocessing_BLP.arguments(
+        SecondInputParametersSchema(unknown=EXCLUDE), location="form"
+    )
+    @PDPreprocessing_BLP.response(HTTPStatus.OK, TaskResponseSchema())
+    @PDPreprocessing_BLP.require_jwt("jwt", optional=True)
+    def post(self, arguments, db_id: int, step_id: int):
+        """Start the calculation task."""
+        db_task: Optional[ProcessingTask] = ProcessingTask.get_by_id(id_=db_id)
+        if db_task is None:
+            msg = f"Could not load task data with id {db_id} to read parameters!"
+            TASK_LOGGER.error(msg)
+            raise KeyError(msg)
+
+        db_task.parameters = SecondInputParametersSchema().dumps(arguments)
+
+        # next step
+        next_step_id = str(step_id + 1)
+        href = url_for(
+            f"{PDPreprocessing_BLP.name}.FinalProcessView",
+            db_id=db_task.id,
+            step_id=next_step_id,
+            _external=True,
+        )
+        ui_href = url_for(
+            f"{PDPreprocessing_BLP.name}.SecondMicroFrontend",
+            db_id=db_task.id,
+            step_id=next_step_id,
+            _external=True,
+        )
+
+        db_task.clear_previous_step()
+        db_task.save(commit=True)
+
+        # all tasks need to know about db id to load the db entry
+        task: chain = second_task.s(db_id=db_task.id, step_id=step_id) | add_step.s(
+            db_id=db_task.id,
+            step_id=next_step_id,
+            href=href,
+            ui_href=ui_href,
+            prog_value=50,
```

Comment:
You could try to calculate a more usable progress value (i.e. `(nr_of_steps / (nr_of_steps + 1)) * 100`)
Or use a progress mode that is not based on percentages (may not be supported by the add_step task).